### ISSUE TITLE
In 1.0.6, relative to 0.23.2, the effect of get_object_importance on the forecast has changed significantly (worse, not better) - object_importance_tutorial.ipynb

### ISSUE THREAD

**Issue by dbulgakov7** [catboost/catboost#2100]
Problem:

Removing rows from the training dataset made the forecast worse, not better. Is this behavior normal? Is get_object_importance still working correctly?

catboost version: 1.0.6
Operating System: Google Colab (python 3.7)
CPU: +
GPU: -

### I have completed the [object_importance_tutorial.ipynb](../blob/master/catboost/tutorials/model_analysis/object_importance_tutorial.ipynb) tutorial in Google Collab with the current version of catboost 1.0.6 and got this result:

||CatBoost 1.0.6, RMSE on the validation dataset|
|-|-|
|Clear train dataset: | 0.2040173285755763|
|Noisy train dataset: | 0.23455697565826905|
|Purified train dataset: | 0.26158053082097327|

RMSE on validation datset when 0 harmful objects from train are dropped: **0.23455697565826905**
RMSE on validation datset when 250 harmful objects from train are dropped: 0.23077052675959442
RMSE on validation datset when 500 harmful objects from train are dropped: 0.22892173282525313
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2325048656521516
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.23616281948796095
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.2413426477428763
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.24430962707884527
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.25479190475958297
RMSE on validation datset when 2000 harmful objects from train are dropped: **0.26158053082097327**

### If you run [object_importance_tutorial.ipynb](../blob/master/catboost/tutorials/model_analysis/object_importance_tutorial.ipynb) with version 0.23.2 from August 2020 (the tutorial was published in February 2020), we will get this result:

||CatBoost 0.23.2, RMSE on the validation dataset|
|-|-|
|Clear train dataset: | 0.20374203709569602|
|Noisy train dataset: | 0.2546074604439433|
|Purified train dataset: | 0.20995709327319195|

RMSE on validation datset when 0 harmful objects from train are dropped: **0.2546074604439433**
RMSE on validation datset when 250 harmful objects from train are dropped: 0.2590571387906804
RMSE on validation datset when 500 harmful objects from train are dropped: 0.3164207507199551
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2472704984365404
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.23881088274295856
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22859104189841842
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.2228825517718825
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.2135612495563327
RMSE on validation datset when 2000 harmful objects from train are dropped: **0.20995709327319195**

**Comment by dbulgakov7** [comment]
Something broke in version 1.0.0 for the CPU model.

<details>
  <summary>Details 0.24.4 - 1.0.6</summary>
  
0.24.4 CPU
clear 0.20374203709569602

RMSE on validation datset when 0 harmful objects from train are dropped: 0.2546074604439433
RMSE on validation datset when 250 harmful objects from train are dropped: 0.2590571387906804
RMSE on validation datset when 500 harmful objects from train are dropped: 0.3164207507199551
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2472704984365404
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.23881088274295856
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22859104189841842
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.2228825517718825
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.2135612495563327
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.20995709327319195

0.24.4 GPU
clear 0.2043093161561304

RMSE on validation datset when 0 harmful objects from train are dropped: 0.23581201969444315
RMSE on validation datset when 250 harmful objects from train are dropped: 0.2326477282826943
RMSE on validation datset when 500 harmful objects from train are dropped: 0.23091096833187075
RMSE on validation datset when 750 harmful objects from train are dropped: 0.22841720794304843
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.22875952277232267
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22948418770862272
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.22842965016744468
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.22593716761032917
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.22862975024227958

---

0.26 CPU
clear 0.20374203709569602

RMSE on validation datset when 0 harmful objects from train are dropped: 0.2546074604439433
RMSE on validation datset when 250 harmful objects from train are dropped: 0.2590571387906804
RMSE on validation datset when 500 harmful objects from train are dropped: 0.3164207507199551
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2472704984365404
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.23881088274295856
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22859104189841842
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.2228825517718825
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.2135612495563327
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.20995709327319195

0.26 GPU
catboost crashes #1735

---

0.26.1 CPU
clear 0.20374203709569602

RMSE on validation datset when 0 harmful objects from train are dropped: 0.2546074604439433
RMSE on validation datset when 250 harmful objects from train are dropped: 0.2590571387906804
RMSE on validation datset when 500 harmful objects from train are dropped: 0.3164207507199551
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2472704984365404
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.23881088274295856
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22859104189841842
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.2228825517718825
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.2135612495563327
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.20995709327319195

0.26.1 GPU
clear 0.2024935436789298

RMSE on validation datset when 0 harmful objects from train are dropped: 0.23492137115037504
RMSE on validation datset when 250 harmful objects from train are dropped: 0.23186795135463548
RMSE on validation datset when 500 harmful objects from train are dropped: 0.22788778369085336
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2275527181092034
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.22467138155938773
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22589384278293703
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.22271966762020087
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.2210866571227945
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.22395811887137518

---

1.0.0 CPU
0.2040173285755763

RMSE on validation datset when 0 harmful objects from train are dropped: 0.23455697565826905
RMSE on validation datset when 250 harmful objects from train are dropped: 0.23077052675959442
RMSE on validation datset when 500 harmful objects from train are dropped: 0.22892173282525313
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2325048656521516
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.23616281948796095
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.2413426477428763
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.24430962707884527
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.25479190475958297
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.26158053082097327

1.0.0 GPU
clear 0.20376530104093035

RMSE on validation datset when 0 harmful objects from train are dropped: 0.23400998959166236
RMSE on validation datset when 250 harmful objects from train are dropped: 0.23139783140732906
RMSE on validation datset when 500 harmful objects from train are dropped: 0.23049212319793433
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2271777051821322
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.22625123000057976
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22524632563400868
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.22419450699311128
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.22521025505840422
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.2235989297176115

---

1.0.6 CPU
clear 0.2040173285755763

RMSE on validation datset when 0 harmful objects from train are dropped: 0.23455697565826905
RMSE on validation datset when 250 harmful objects from train are dropped: 0.23077052675959442
RMSE on validation datset when 500 harmful objects from train are dropped: 0.22892173282525313
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2325048656521516
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.23616281948796095
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.2413426477428763
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.24430962707884527
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.25479190475958297
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.26158053082097327

1.0.6 GPU
clear 0.20376530104093035

RMSE on validation datset when 0 harmful objects from train are dropped: 0.23400998959166236
RMSE on validation datset when 250 harmful objects from train are dropped: 0.23139783140732906
RMSE on validation datset when 500 harmful objects from train are dropped: 0.23049212319793433
RMSE on validation datset when 750 harmful objects from train are dropped: 0.2271777051821322
RMSE on validation datset when 1000 harmful objects from train are dropped: 0.22625123000057976
RMSE on validation datset when 1250 harmful objects from train are dropped: 0.22524632563400868
RMSE on validation datset when 1500 harmful objects from train are dropped: 0.22419450699311128
RMSE on validation datset when 1750 harmful objects from train are dropped: 0.22521025505840422
RMSE on validation datset when 2000 harmful objects from train are dropped: 0.2235989297176115  
</details>

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
all is korrect except the tutorial 🙂 
by default, catboost uses a high learning rate
the models get over-fitted too early, and we look at irrelevant values of rmse
just add `'learning_rate': 0.001` to the catboost parameters
i will update the tutorial
